dids/src/main/kotlin/web5/sdk/dids/did/BearerDid.kt

```diff
@@ -0,0 +1,153 @@
+   * This function also returns the verification method needed to verify the signature.
+   *
+   * Providing the verification method allows the caller to provide the signature's recipient
+   * with a reference to the verification method needed to verify the payload. This is often done
+   * by including the verification method id either alongside the signature or as part of the header
+   * in the case of JSON Web Signatures.
+   *
+   * The verifier can dereference the verification method id to obtain the public key needed to verify the signature.
+   *
+   * This function takes a Verification Method selector that can be used to select a specific verification method
+   * from the DID Document if desired. If no selector is provided, the payload will be signed with the key associated
+   * to the first verification method in the DID Document.
+   *
+   * The selector can either be a Verification Method ID or a Purpose. If a Purpose is provided, the first verification
+   * method in the DID Document that has the provided purpose will be used to sign the payload.
+   *
+   * The returned signer is a function that takes a byte payload and returns a byte signature.
+   */
+  @JvmOverloads
+  public fun getSigner(selector: VMSelector? = null): Pair<DidSigner, VerificationMethod> {
+    val verificationMethod = document.selectVerificationMethod(selector)
+
+    val kid = verificationMethod.publicKeyJwk?.computeThumbprint()
+      ?: throw Exception("Failed to compute key alias")
+
+    val signer: DidSigner = { payload ->
+      keyManager.sign(kid, payload)
+    }
+
+    return Pair(signer, verificationMethod)
+  }
+
+  /**
+   * Converts a `BearerDid` object to a portable format containing the URI and verification methods
+   * associated with the DID.
+   *
+   * This method is useful when you need to represent the key material and metadata associated with
+   * a DID in format that can be used independently of the specific DID method implementation. It
+   * extracts both public and private keys from the DID's key manager and organizes them into a
+   * `PortableDid` structure.
+   *
+   * @returns A `PortableDid` containing the URI, DID document, metadata, and optionally private
+   *          keys associated with the `BearerDid`.
+   */
+  public fun export(): PortableDid {
+
+    check(keyManager is KeyExporter) {
+      "KeyManager must implement KeyExporter to export keys"
+    }
+
+    val keyExporter = keyManager as KeyExporter
+    val privateKeys = mutableListOf<Jwk>()
+
+    document.verificationMethod?.forEach { vm ->
+      val keyAliasResult = runCatching { vm.publicKeyJwk?.computeThumbprint() }
+      if (keyAliasResult.isSuccess) {
+        val keyAlias = keyAliasResult.getOrNull()
+        keyExporter.exportKey(keyAlias!!.toString()).let { key ->
+          privateKeys.add(key)
+        }
+      }
+    }
+
+    return PortableDid(
```

Comment:
is there a unit test that makes sure the output of this portable did matches the js and/or golang one?

Reply:
hmm... that sounds like a test vector job. i shall write a test vector!

Reply:
wrote test vector and a test against it: https://github.com/TBD54566975/web5-kt/pull/262/commits/b47299053369c359cb49aad2b2ac6612f36d3419